File: src/main/java/corp/phonebook/config/CookieService.java
package corp.phonebook.config;

import org.springframework.stereotype.Service;
import jakarta.servlet.http.Cookie;
import jakarta.servlet.http.HttpServletResponse;

import java.net.URLEncoder;
import java.nio.charset.StandardCharsets;
import java.time.LocalDateTime;
import java.time.format.DateTimeFormatter;

@Service
public class CookieService {

    public void addSessionCookie(HttpServletResponse response, String sessionId) {
        Cookie sessionCookie = new Cookie("SESSION_ID", sessionId);
        sessionCookie.setHttpOnly(true);
        sessionCookie.setPath("/");
        sessionCookie.setMaxAge(60 * 20);

        response.addCookie(sessionCookie);

        LocalDateTime expirationTime = LocalDateTime.now().plusMinutes(20);
        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm:ss");

        String formattedExpirationTime = expirationTime.format(formatter);
        formattedExpirationTime = URLEncoder.encode(formattedExpirationTime, StandardCharsets.UTF_8);

        Cookie expirationCookie = new Cookie("EXPIRED_COOKIE", formattedExpirationTime);
        expirationCookie.setHttpOnly(true);
        expirationCookie.setPath("/");
        expirationCookie.setMaxAge(60 * 20);

        response.addCookie(expirationCookie);
    }

    public void deleteSessionCookie(HttpServletResponse response) {
        Cookie logoutSessionCookie = new Cookie("SESSION_ID", null);
        logoutSessionCookie.setHttpOnly(true);
        logoutSessionCookie.setPath("/");
        logoutSessionCookie.setMaxAge(0);

        Cookie logoutTimeCookie = new Cookie("EXPIRED_COOKIE", null);
        logoutTimeCookie.setHttpOnly(true);
        logoutTimeCookie.setPath("/");
        logoutTimeCookie.setMaxAge(0);

        response.addCookie(logoutTimeCookie);
    }
}

File: src/main/kotlin/corp/phonebook/config/CookieService.kt
package corp.phonebook.config

import org.springframework.stereotype.Service
import jakarta.servlet.http.Cookie
import jakarta.servlet.http.HttpServletResponse
import java.net.URLEncoder
import java.nio.charset.StandardCharsets
import java.time.LocalDateTime
import java.time.format.DateTimeFormatter

@Service
class CookieService {
    fun addSessionCookie(response: HttpServletResponse, sessionId: String) {
        val sessionCookie = Cookie("SESSION_ID", sessionId).apply {
            isHttpOnly = true
            path = "/"
            maxAge = 60 * 20
        }
        response.addCookie(sessionCookie)

        val expirationTime = LocalDateTime.now().plusMinutes(20)
        val formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm:ss")
        val formattedExpirationTime = URLEncoder.encode(expirationTime.format(formatter), StandardCharsets.UTF_8)

        val expirationCookie = Cookie("EXPIRED_COOKIE", formattedExpirationTime).apply {
            isHttpOnly = true
            path = "/"
            maxAge = 60 * 20
        }
        response.addCookie(expirationCookie)
    }

    fun deleteSessionCookie(response: HttpServletResponse) {
        val logoutSessionCookie = Cookie("SESSION_ID", null).apply {
            isHttpOnly = true
            path = "/"
            maxAge = 0
        }
        val logoutTimeCookie = Cookie("EXPIRED_COOKIE", null).apply {
            isHttpOnly = true
            path = "/"
            maxAge = 0
        }

        response.addCookie(logoutSessionCookie)
        response.addCookie(logoutTimeCookie)
    }
}